lib/validator.js

```diff
@@ -15,48 +54,131 @@ var inMapping = {
  * to JSON-schema for a request
  *
  * @param parameters list of params
- * @returns JSON schema
+ * @returns {Object} JSON schema
+ * @private
  */
-function convertToJsonSchema(parameters) {
+Validator.prototype._convertToJsonSchema = function(parameters) {
     var schema = {
         type: 'object',
         properties: {}
     };
 
     parameters.forEach(function(param) {
-        if (!schema.properties[inMapping[param.in]]) {
-            schema.properties[inMapping[param.in]] = {
-                type: 'object',
-                properties: {}
-            };
-            // 'required' array must have at least one element according to json-schema spec,
-            // se we can't preinitialize it.
-            schema.required = schema.required || [];
-            schema.required.push(inMapping[param.in]);
-        }
-        var reqPartSchema = schema.properties[inMapping[param.in]];
-        reqPartSchema.properties[param.name] = {};
-        if (param.required) {
-            reqPartSchema.required = reqPartSchema.required || [];
-            reqPartSchema.required.push(param.name);
+        if (param.in !== 'body') {
+            if (!schema.properties[inMapping[param.in]]) {
+                schema.properties[inMapping[param.in]] = {
+                    type: 'object',
+                    properties: {}
+                };
+                // 'required' array must have at least one element according to json-schema spec,
+                // se we can't preinitialize it.
+                schema.required = schema.required || [];
+                if (schema.required.indexOf(inMapping[param.in]) < 0) {
+                    schema.required.push(inMapping[param.in]);
+                }
+            }
+
+            var reqPartSchema = schema.properties[inMapping[param.in]];
+            var paramSchema = { type: param.type };
+            supportedValidators.forEach(function(validator) {
+                paramSchema[validator] = param[validator];
+            });
+            reqPartSchema.properties[param.name] = paramSchema;
+            if (param.required) {
+                reqPartSchema.required = reqPartSchema.required || [];
+                reqPartSchema.required.push(param.name);
+            }
+        } else {
```

Comment:
According to the swagger specification, you can have only one `body` parameter - it represents the whole body, not some field in the body. And I assume our spec are valid from swagger perspective.